src/services/importService.ts

```diff
@@ -0,0 +1,191 @@
+            },
+        };
+
+        const connections: IConnection[] = [sourceConnection, targetConnection];
+
+        return(connections);
+    }
+
+    /**
+     * Parse v1 project's tag string and return array of ITags
+     * @param project - V1 Project Content and File Information
+     */
+    private parseTags(project: any): ITag[] {
+        const finalTags: ITag[] = [];
+        const tagStrings = project.inputTags.split(",");
+        const tagColors = project.tag_colors;
+
+        for (let i = 0; i < tagColors.length; i++) {
+            const newTag = {
+                name: tagStrings[i],
+                color: tagColors[i],
+            };
+            finalTags.push(newTag);
+        }
+        return finalTags;
+    }
+
+    /**
+     * Generate assets based on V1 Project frames and regions
+     * @param project - V1 Project Content and File Information
+     */
+    private generateAssets(project: any, assetService: AssetService): Array<Promise<IAssetMetadata>> {
+        let originalProject: IV1Project;
+        const generatedAssetMetadata: Array<Promise<IAssetMetadata>> = [];
+        let generatedRegion: IRegion;
+        let assetState: AssetState;
+        const currentTagColorIndex = randomIntInRange(0, TagColors.length);
+
+        originalProject = JSON.parse(project.content);
+
+        for (const frameName in originalProject.frames) {
+            if (originalProject.frames.hasOwnProperty(frameName)) {
+                const frameRegions = originalProject.frames[frameName];
+                const asset = AssetService.createAssetFromFilePath(
+                    `file:${project.file.path.replace(/[^\/]*$/, "")}${frameName}`);
+                const assetMetadata = assetService.getAssetMetadata(asset).then((metadata) => {
+                    // assetMetadata = metadata;
+                    assetState = originalProject.visitedFrames.indexOf(frameName) > -1 && frameRegions.length > 0
+                                ? AssetState.Tagged : (originalProject.visitedFrames.indexOf(frameName) > -1
+                                ? AssetState.Visited : AssetState.NotVisited);
+                    metadata.asset.state = assetState;
+
+                    /*
+                    for (const region of frameRegions) {
+                        generatedRegion = canvasHelpers.getRegion(region, "rect", region.UID)
+                        generatedRegion.tags = region.tags.map((tag) => {
+                                let newTag: ITag;
+                                newTag = {
+                                    name: tag,
+                                    color: TagColors[(currentTagColorIndex + 1) % TagColors.length],
+                                };
+                                return newTag;
+                            });
+                    */
+
+                    for (const region of frameRegions) {
+                        generatedRegion = {
```

Comment:
Looks like `generatedRegion` should be scoped to within this loop vs. being scoped outside of this loop - otherwise you are redefining the region on every iteration.